### DISCUSSION TITLE
How to transfer strain in spherical coordinates to Cartesian coordinate?

### DISCUSSION THREAD

**Discussion by RECHOA** [idaholab/moose#22105] [Q&A Modules: Solid mechanics] [answered]
Hi all,
If i know the raidal and hoop strain of a material in spherical coordinates as folliws:
 ε<sub>r</sub>= _-0.007T_ ;  ε<sub>t</sub>= _-0.004T_ ; _T_ is the tempearature.

Now i want to analyze its mechanical behaviour by using MOOSE in the Cartesian coordinate (coord-type=XYZ), how should I build the strain tensor **ε** in EigenStrain.C?

**Comment by GiudGiud** [comment, marked as answer]
@recuero 
I think you ll want to do the math, build a generic constant tensor, and use that in the calculation.
I doubt we have a utility for you here but Antonio will know

**Reply by recuero** [reply]
I doubt we have a utility for that too. It seems that once you perform the transformation from your spherical coordinates to the Cartesian systems, you'll end up with a nontrivial dependency on the temperature. I wonder if a simple way forward can be just to inherit from `ComputeEigenstrain` and perform the transformation there pulling the temperature variable.

Maybe @jiangwen84 knows if there is a simpler way to do that.

**Reply by hugary1995** [reply]
The general steps are:
1. Calculate the eigen deformation gradient from the eigen strain, IIRC tensor mechanics uses the log strain measure.
2. Calculate the scalar-valued thermal expansion, i.e. the determinant of the eigen deformation gradient, which is convenient as it is an invariant.
3. Calculate the eigen deformation gradient in Cartesian from the calculated thermal expansion.
4. Calculate the eigen strain from the eigen deformation gradient in Cartesian.

Your eigen strain might be trickier as it seems to be anisotropic. But the general steps should apply, you just do it for the two directions and add the eigen strains.

**Reply by hugary1995** [reply]
And of course you can do it as @recuero suggested -- directly transform the eigen strain tensor using a rotation tensor. It will require users to define the axisymmetric RZ coordinate system, e.g. the symmetry axis.

**Reply by RECHOA** [reply]
Thanks for your reply. I am sorry i didn't understand the general steps. Are there articles or formula derivation i can refer to? 

As for the other method (directly transforming the eigen strain tensor using a rotation tensor), If I have multiple spheres like this to simulate, does that mean I have to set up multiple spherical coordinates and convert the strain to cartesian coordinates in each spherical coordinate？

**Reply by hugary1995** [reply]
> Are there articles or formula derivation i can refer to?

I'm not aware of any unfortunately.

> As for the other method (directly transforming the eigen strain tensor using a rotation tensor), If I have multiple spheres like this to simulate, does that mean I have to set up multiple spherical coordinates and convert the strain to cartesian coordinates in each spherical coordinate？

Yes. Intuitively that's how it works if I understand your problem description correctly: you know how the spheres are going to expand radially and axially in their respective spherical coordinates, so they can't share a common spherical coordinate system as they are transformed into Cartesian.

**Reply by RECHOA** [reply]
> you know how the spheres are going to expand radially and axially in their respective spherical coordinates, so they can't share a common spherical coordinate system as they are transformed into Cartesian.

yes, the expansion of the spheres are  konwn in in their respective spherical coordinates. In each sphere it follws ε<sub>r</sub>= -0.007T ; ε<sub>t</sub>= -0.004T, i.e. the strain tensor in spherical coordinate is known as
### S<sub>_sphere_</sub> =
 ![matrix](https://user-images.githubusercontent.com/33534979/190587491-599afe86-8a8f-4ba7-8288-76d3bd1512b2.JPG)
So now i need to find a way (maybe a transform matrix **M**) to convert  **S<sub>_sphere_</sub>** in  respective spherical coordinates to **S<sub>_xyz_</sub>** in Cartesian coordinate system
